src/OrleansTestingHost/TestingSiloHost.cs

```diff
@@ -403,12 +401,13 @@ public SiloHandle RestartSilo(SiloHandle instance)
                 {
                     Primary = newInstance;
                 }
-                else if (type == Silo.SiloType.Secondary)
+                else if (Secondary == instance)
```

Comment:
Sure, done on #1583 now. As FYI, I tracked the introduction of this bug to #1255, which is one of the multi-part migration of the tests from VSO, which correspond roughly to the timeframe that our functionals became unstable.